js_modules/dagster-ui/packages/ui-core/src/runs/RunConfigDialog.tsx

```diff
@@ -73,27 +72,26 @@ export const RunConfigDialog = ({run, isJob}: {run: RunFragment; isJob: boolean}
     }
   };
 
-  const jobPath = workspacePathFromRunDetails({
-    id: run.id,
+  const jobLink = workspacePipelineLinkForRun({
+    run,
     repositoryName: run.repositoryOrigin?.repositoryName,
     repositoryLocationName: run.repositoryOrigin?.repositoryLocationName,
-    pipelineName: run.pipelineName,
     isJob,
   });
 
   return (
     <div>
       <Group direction="row" spacing={8}>
-        {run.hasReExecutePermission ? (
-          <AnchorButton icon={<Icon name="edit" />} to={jobPath}>
-            Open in Launchpad
-          </AnchorButton>
-        ) : (
-          <Tooltip content={NO_LAUNCH_PERMISSION_MESSAGE} useDisabledButtonTooltipFix>
+        {jobLink.disabledReason ? (
+          <Tooltip content={jobLink.disabledReason} useDisabledButtonTooltipFix>
             <Button icon={<Icon name="edit" />} disabled>
```

Comment:
Worth adjusting the `icon` here, since the action is slightly different in each case now?

Reply:
Ahh yeah I think that's a good idea 👍